tests/test_generic/tests/test_industry_requirements.py

```diff
@@ -59,3 +73,20 @@ def test_cloc_exclude_view(self):
                 message = "The view '%s' is counted in the maintenance lines. " % cloc_entry
                 message += "Please add a '__cloc_exclude__' entry in 'ir_model_data'."
                 self.assertEqual(len(cloc_entry.split(':')), 2, message)
+
+    def test_sale_ok_and_is_published_in_db(self):
+        models = ["product.template", "product.product"]
+        for model in models:
+            if "is_published" in self.env[model]._fields and "sale_ok" in self.env[model]._fields:
+                records = self.env[model].search(
+                    [("is_published", "=", True), ("sale_ok", "=", False)]
+                )
+                if records:
+                    record_info = ", ".join(
+                        ["%s (ID: %s)" % (record.display_name, record.id) for record in records]
+                    )
+                    self.assertFalse(
```

Comment:
Well, assertFalse after the if is useless. Replace the if by the assert and don't assign the info to an intermediate variable.